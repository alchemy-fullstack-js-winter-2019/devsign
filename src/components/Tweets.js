import React from 'react';
import PropTypes from 'prop-types';
import Tweet from './Tweet';

function Tweets({ tweets }) {
  const tweetList = tweets.map((tweet, i) => {
    return <li key={i}><Tweet
      profileImage={tweet.profileImage}
      name={tweet.name} 
      text={tweet.text}
      handle={tweet.handle}
    /></li>;
  });
  return (
    <section>
      <ul>
        {tweetList}
      </ul>
    </section>
  );
}

Tweets.propTypes = {
  tweets: PropTypes.array.isRequired
};

export default Tweets;
