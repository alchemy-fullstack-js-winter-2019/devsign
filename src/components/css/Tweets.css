.tweetItems {
  list-style-type: none;
}