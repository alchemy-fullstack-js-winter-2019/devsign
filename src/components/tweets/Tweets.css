.Tweets {
  list-style: none;
  
  
  li {
    border: 2px solid black;
    margin: 2vh 2vw;
    width: 70vw;
  }
}