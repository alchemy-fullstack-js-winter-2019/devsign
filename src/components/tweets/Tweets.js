import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Tweet from './Tweet';

function Tweets({ tweets }) {
  const tweetsList = tweets.map(tweet =>
    <li key={tweet.id}>
      <Link to={`/user/${tweet.userId}`}>
        <Tweet tweet={tweet} />
      </Link>
    </li>
  );

  return (
    <>
      <h2>Latest Tweets:</h2>
      <ul>{tweetsList}</ul>
    </>
  );
}

Tweets.propTypes = {
  tweets: PropTypes.array.isRequired
};

export default Tweets;
