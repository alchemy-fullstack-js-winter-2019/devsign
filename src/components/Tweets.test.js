import React from 'react';
import renderer from 'react-test-renderer';
import Tweets from './Tweets';

describe('Tweets Test', () => {
  it('mathes a snapshot', () => {
    const tweets = [
      {
        name: 'BugS',
        profileImage: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSZRJRA8ywW_omvM7B-9KwlPGxsP6WCdGuKShl211WtKEoJ_h6D',
        handle: '@bugsbunny',
        text: 'It is sunny'
      },
      {
        name: 'John Gomex',
        profileImage: 'http://johngomex.jpeg',
        handle: '@JohnX',
        text: 'It is fake news'
      }
    ];
    const tree = renderer.create(
      <Tweets tweets={tweets} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
