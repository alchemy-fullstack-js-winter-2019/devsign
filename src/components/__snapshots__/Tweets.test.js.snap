// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweets Test mathes a snapshot 1`] = `
<section>
  <ul>
    <li>
      tweets
    </li>
  </ul>
</section>
`;
